docs/src/integrate/integration-tutorial.md

```diff
@@ -0,0 +1,230 @@
+
+```javascript
+// example-eslint-integration.js
+
+// Put previous functions all together
+async function lintFiles(filePaths) {
+
+    // The ESLint configuration. Alternatively, you could load the configuration
+    // from a .eslintrc file or just use the default config.
+    const overrideConfig = {
+        env: {
+            es6: true,
+            node: true,
+        },
+        parserOptions: {
+            ecmaVersion: 2018,
+        },
+        rules: {
+            "no-console": "error",
+            "no-unused-vars": "warn",
+        },
+    };
+
+    const eslint = createESLintInstance(overrideConfig);
+    const results = await lintAndFix(eslint, filePaths);
+    return outputLintingResults(results);
+}
+
+// Export integration
+module.exports = { lintFiles }
+```
+
+Here's the complete code example for `example-eslint-integration.js`:
+
+```javascript
+const { ESLint } = require("eslint");
+
+// Create an instance of ESLint with the configuration passed to the function
+function createESLintInstance(overrideConfig){
+  return new ESLint({ useEslintrc: false, overrideConfig: overrideConfig });
+}
+
+// Lint the specified files and return the error results
+async function lintAndFix(eslint, filePaths) {
+  const results = await eslint.lintFiles(filePaths);
+
+  // Apply automatic fixes and output fixed code
+  await ESLint.outputFixes(results);
+
+  return results;
+}
+
+// Log results to console if there are any problems
+function outputLintingResults(results) {
+  if (results.length) {
+    console.log("Linting errors found!");
+    console.log(results);
+  } else {
+    console.log("No linting errors found.");
+  }
+  return results;
+}
+
+// Put previous functions all together
+async function lintFiles(filePaths) {
+
+    // The ESLint configuration. Alternatively, you could load the configuration
+    // from a .eslintrc file or just use the default config.
+    const overrideConfig = {
+        env: {
+            es6: true,
+            node: true,
+        },
+        parserOptions: {
+            ecmaVersion: 2018,
+        },
+        rules: {
+            "no-console": "error",
+            "no-unused-vars": "warn",
+        },
+    };
+
+    const eslint = createESLintInstance(overrideConfig);
+    const results = await lintAndFix(eslint, filePaths);
+    return outputLintingResults(results);
+}
+
+// Export integration
+module.exports = { lintFiles }
+```
+
+## Conclusion
+
+In this tutorial, we have covered the essentials of using the `ESLint` class to lint files and retrieve results in your projects. This knowledge can be applied to create custom integrations, such as code editor plugins, to provide real-time feedback on code quality.
+
+## View the Tutorial Code
+
+You can view the annotated source code for the tutorial [here](https://github.com/eslint/eslint/tree/main/docs/_integration-tutorial-code).
```

Comment:
Rather than creating a top-level directory under `docs`, maybe we should have `docs/_examples` for all examples and then a subdirectory under `docs/_examples` for the integration example code?

Reply:
moved